capa/features/extractors/ghidra/function.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (C) 2020 Mandiant, Inc. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at: [package root]/LICENSE.txt
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+#  is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+from typing import Tuple, Iterator
+
+import ghidra
+from ghidra.program.model.block import BasicBlockModel
+
+import capa.features.extractors.ghidra.helpers
+from capa.features.common import Feature, Characteristic
+from capa.features.address import Address, AbsoluteVirtualAddress
+from capa.features.extractors import loops
+from capa.features.extractors.base_extractor import FunctionHandle
+
+currentProgram: ghidra.program.database.ProgramDB
+monitor = getMonitor()  # type: ignore [name-defined]
+
+
+def extract_function_calls_to(fh: ghidra.program.database.function.FunctionDB):
+    """extract callers to a function"""
+    for ref in fh.getSymbol().getReferences():
+        if ref.getReferenceType().isCall():
+            yield Characteristic("calls to"), AbsoluteVirtualAddress(ref.getFromAddress().getOffset())
+
+
+def extract_function_loop(fh: ghidra.program.database.function.FunctionDB):
+    edges = []
+    model = BasicBlockModel(currentProgram)  # does not allow overlap, so we have to iterate ourself
+    addr_set = fh.getBody()
+
+    for block in model.getCodeBlocksContaining(addr_set, monitor):
+        dests = block.getDestinations(monitor)
+        s_addrs = block.getStartAddresses()
+
+        while dests.hasNext():  # Python error forces us to use iterator functions
```

Comment:
I'm thinking this may be on the Ghidrathon side. For the most part, when the Ghidra API functions return an iterator, we can access it on the python side with the regular for loops; however, for this particular function `getDestinations()` it throws a type error when we try to iterate with a classic for loop. 

The only way I've gotten it to work is with the `while dests.hasNest()` loop.

<img width="658" alt="image" src="https://github.com/mandiant/capa/assets/66766340/154e15e8-e052-4824-ade3-f396bed9ba57">

Not exactly sure why this iterator in particular is treated differently than the others we've used so far.